src/Components/InventoryPage/InventoryPage.tsx

```diff
@@ -0,0 +1,196 @@
+
+  useEffect(() => {
+    dispatch({ type: 'INVENTORY_INIT' });
+    dispatch({ type: 'RESET_PAGE' });
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //@ts-ignore
+    inventory?.current?.onRefreshData();
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch({
+      type: 'SELECT_ENTITIES',
+      payload: {
+        selected: selectedIds,
+      },
+    });
+  }, [selectedIds]);
+
+  const onComplete = (result, filtersString: string) => {
+    setTotal(result.total);
+    setItems(result.results);
+    setActiveFiltersString(filtersString);
+  };
+
+  const getEntities = useGetEntities(onComplete, {
+    selectedIds,
+  });
+
+  const mergedColumns = (defaultColumns) =>
+    systemColumns(chrome?.isBeta?.()).map((column: SystemColumn) => {
+      const defaultColumn = defaultColumns.find((defaultCol) => defaultCol.key === column.key);
+      return {
+        ...defaultColumn,
+        ...column,
+        props: {
+          ...defaultColumn?.props,
+          ...column?.props,
+        },
+      };
+    });
+
+  const selectIds = (_event, _isSelected, _index, entity) => {
+    const newSelectedIds: string[] = [...selectedIds];
+
+    !newSelectedIds.includes(entity.id) ? newSelectedIds.push(entity.id) : newSelectedIds.splice(newSelectedIds.indexOf(entity.id), 1);
+
+    setSelectedIds(newSelectedIds);
+  };
+
+  const bulkSelectIds = async (type: string, options?) => {
+    const newSelectedIds = [...selectedIds];
+
+    setBulkLoading(true);
+    switch (type) {
+      case 'none': {
+        setSelectedIds([]);
+        break;
+      }
+
+      case 'page': {
+        options.items.forEach((item) => {
+          if (!newSelectedIds.includes(item.id)) {
+            newSelectedIds.push(item.id);
+          }
+        });
+
+        setSelectedIds(newSelectedIds);
+        break;
+      }
+
+      case 'all': {
+        const ids = [] as string[];
+        const perPage = 100;
+        for (let page = 1; page <= Math.ceil(total / perPage); page++) {
+          const { results: pageResults } = await fetchSystems(`?per_page=100&page=${page}${activeFiltersString}`);
+          ids.push(...pageResults.map(({ id }) => id));
+        }
+        setSelectedIds(ids);
+        break;
+      }
+    }
+    setBulkLoading(false);
+  };
+
+  return isLoadingInventory ? (
+    <Bullseye>
+      <Spinner />
+    </Bullseye>
+  ) : canReadHostsInventory ? (
+    <InventoryTable
+      isFullView
+      autoRefresh
+      showTags
+      hideFilters={{
+        all: true,
+        name: false,
+        operatingSystem: false,
+      }}
+      columns={mergedColumns}
+      ref={inventory}
+      fallback={
+        <Bullseye>
+          <Spinner />
+        </Bullseye>
+      }
+      onLoad={defaultOnLoad(getRegistry)}
+      getEntities={getEntities}
+      tableProps={{
+        isStickyHeader: true,
+        isStriped: true,
+        canSelectAll: false,
+        onSelect: items.length ? selectIds : null,
```

Comment:
I noticed there is some difference between our table and inventory, if you click checkbox to select all, it does many... and I mean many requests to select all 4k systems (when I am logged on my profile) but inventory table selects only the first page.

![image](https://github.com/RedHatInsights/in-place-upgrades-frontend/assets/19702477/4b81802b-e0c6-47f7-bb89-8e35936b4795)

Was this intentional?

Reply:
Nope, I will change it to page select :+1: